mne/preprocessing/ica.py

```diff
@@ -1451,10 +1462,8 @@ def _apply_evoked(self, evoked, include, exclude, n_pca_components):
 
     def _pick_sources(self, data, include, exclude):
         """Aux function."""
-        if exclude is None:
-            exclude = self.exclude
-        else:
-            exclude = list(set(self.exclude + list(exclude)))
+        # We assume exclude = self._check_exclude(exclude)  has already been
```

Comment:
is this assumption safe? Or in other words: if self._check_exclude is always called by mne code before this one, we don't have to assume anything here. :) (this is a private function after all)

Reply:
I turned the comment into the required line of code and removed that in the calling functions, thereby removing duplications. OK?

Reply:
Yes, that should be ok, we'll say what tests have to say about that. :)